src/sentry/api/serializers/models/alert_rule.py

```diff
@@ -118,13 +118,23 @@ def get_attrs(self, item_list, user, **kwargs):
                     "originalAlertRuleId"
                 ] = activity.previous_alert_rule_id
 
+        if "latestIncident" in self.expand:
+            incident_map = {}
+            if "latestIncident" in self.expand:
+                for incident in Incident.objects.filter(
+                    id__in=[x.incident_id for x in alert_rules.values()]
```

Comment:
Let's try and keep this logic entirely contained in the serializer, rather than rely on `incident_id` being present.

```
Incident.objects.filter(
    id__in=Incident.objects.filter(alert_rule__in=alert_rules).values("alert_rule_id").annotate(incident_id=Max("id")).values("incident_id")
)
```

Reply:
@wedamija  sounds good